### DISCUSSION TITLE
Real Time Validation fails at Input if leaving without value

### DISCUSSION THREAD

**Discussion by SDJeff** [livewire/livewire#5362] [7. Bugs (archived)]
Hi,

i have here an input:

    <x-inputs.maskable mask="####" wire:model.lazy="employees_amount" label="{{ __('Number of employees') }}" corner-hint="* placeholder="{{ __('How many employees does your company have?') }}" />

    public int $employees_amount;
 
    protected $rules = [
        'employees_amount' => 'required|numeric|min:1'
    ];

    public function mount()
    {
        $this->employees_amount = 0;
    }

If you leave the input directly with TAB Key or by mouse click you got an error:

_Cannot assign null to property App\Http\Livewire\Forms\CompanyInformationForm::$employees_amount of type int_

But the field is mounted with 0 as default. On top it doesn't calling the 

`public function updated($propertyName)`

only if you type in at least a number.

**Comment by Martleby** [comment]
As far as I understand it...

The validation only occurs *after* the property has been set. In your case, because you've type hinted the property, PHP is throwing an error because the value is not an int (and the property is not a nullable int).

The easiest way around this is not to type hint the property, and then rely on the updated method to run the validation:

```php
<?php

use Livewire\Component;

class LivewireComponent extends Component
{
    public $employeesAmount;

    protected $rules = [
        'employeesAmount' => 'required|numeric|min:1'
    ];

    public function updated($propertyName)
    {
        $this->validateOnly($propertyName);
    }

    public function mount()
    {
        $this->employeesAmount = 0;
    }
  
    public function render()
    {
        return view('livewire-component');
    }
}
```

and blade:

```blade
<div>
  <input wire:model.lazy="employeesAmount"><br>

   @error('employeesAmount') <span class="error">{{ $message }}</span> @enderror

</div>
```

You might also be able to hook into the updatingEmployeesAmount method to inspect the value before it's assigned. Not sure.

BTW: I would *highly recommend* using PHP naming conventions for your property names: so ```$employeesAmount``` not ```$employees_amount```. Trust me, you'll save yourself hours of frustration.

Cheers,
Martin.

**Reply by SDJeff** [reply]
Ok. I was thinking, that i can solve it with:

`public ?int $employees_amount;`

But here the error is also thrown even if the type hinted field is nullable.

And you are right, this is the name of the field at the database table. Sure i have to use in Laravel CamelCase.
